ui/src/websites/wordpress/WordPress.tsx

```diff
@@ -0,0 +1,196 @@
+import { WebsiteSectionProps } from '../form-sections/website-form-section.interface';
+import GenericFileSubmissionSection from '../generic/GenericFileSubmissionSection';
+import { GenericSelectProps } from '../generic/GenericSelectProps';
+import GenericSubmissionSection from '../generic/GenericSubmissionSection';
+import { WebsiteImpl } from '../website.base';
+import { LoginDialogProps } from '../interfaces/website.interface';
+import WordPressAccountInfo from './WordPressAccountInfo';
+
+export class WordPress extends WebsiteImpl {
+  internalName: string = 'WordPress';
+  name: string = 'WordPress Instance';
+  supportsAdditionalFiles: boolean = false;
+  supportsTags: boolean = false;
+  loginUrl: string = '';
+
+  LoginDialog = (props: LoginDialogProps) => <WordPressAccountInfo {...props} />;
+
+  FileSubmissionForm = (props: WebsiteSectionProps<FileSubmission, WordPressFileOptions>) => (
+    <WordPressFileSubmissionForm
+      key={props.part.accountId}
+      {...props}
+    />
+  );
+
+  NotificationSubmissionForm = (
+    props: WebsiteSectionProps<Submission, WordPressNotificationOptions>
+  ) => (
+    <WordPressNotificationSubmissionForm
+      key={props.part.accountId}
+      {...props}
+    />
+  );
+}
+
+// TODO: implement options
+// slug (string)
+// comment status (open/close)
+// format (standard, aside, chat, gallery, link, image, quote, status, video, audio)
+// categories (string input)
+// status (publish, future, draft, pending [review], private)
+// sticky (boolean)
+class WordPressNotificationSubmissionForm extends GenericSubmissionSection<
+  WordPressNotificationOptions
+> {
+  renderLeftForm(data: WordPressNotificationOptions) {
+    const elements = super.renderLeftForm(data);
+    elements.push(
+      <div>
+      </div>,
+      <Form.Item label="Post slug">
+        <Input value={data.slug} onChange={this.handleValueChange.bind(this, 'slug')} />
+      </Form.Item>,
+
+      <Form.Item label="Post status">
+        <Select
+          {...GenericSelectProps}
+          className="w-full"
+          value={data.status}
+          onSelect={this.setValue.bind(this, 'status')}
+        >
+          <Select.Option value="publish">Publish</Select.Option>
+          <Select.Option value="future">Future</Select.Option>
+          <Select.Option value="draft">Draft</Select.Option>
+          <Select.Option value="pending">Pending Review</Select.Option>
+          <Select.Option value="private">Private</Select.Option>
+        </Select>
+      </Form.Item>,
+
+      <Form.Item label="Post comment status">
+        <Select
+          {...GenericSelectProps}
+          className="w-full"
+          value={data.commentStatus}
+          onSelect={this.setValue.bind(this, 'commentStatus')}
+        >
+          <Select.Option value="open">Open</Select.Option>
+          <Select.Option value="closed">Closed</Select.Option>
+        </Select>
+      </Form.Item>,
+
+      <Form.Item label="Category IDs (comma seperated)">
```

Comment:
q: where would a user get these Ids? Are they dynamic to each wordpress instance and are they available to be retrieved via api call to display in a multi-select dropdown?

Reply:
They are dynamic to each WP instance, you can retrieve them via an API call, but I haven't worked out how to display these on the frontend of postybirb yet

Reply:
Ah, you could look at the fur-affinity.service.ts for an example for pulling things down from api and getting them into the UI.

But the gist is to retrieve the categories through an API call, store them in a this.storeAccountInformation(profileId, propName, object_to_store)

You can check out the generic.controller.ts to see about the call that can be made from the UI to retrieve the propName you set.

Then in the actual UI, you can add to your tsx files a WebsiteService call to `WebsiteService.getAccountInformation(website, propName). Then you can store it in the react component to display however you like.

You can see an example of the UI call made in Tumblr.tsx in the constructors of the react elements.